BoneCentral/Brain/app.js

```diff
@@ -0,0 +1,24 @@
+var Config              = require('./config.json');
+var Ports               = require('./Sockets/Ports.json');
+var Sockets             = require('./Sockets');
+var Peripherals        	= require('./CppInterface/PeripheralsProcess.js');
+var HardwareFactory     = require('./CppInterface/Factory.js');
+var Vision              = require('./VisionInterface');
+
+var server              = Sockets.createSocket(Ports.DispatcherPort);
```

Comment:
I have to disagree. App.js (while improperly named) is a form of "builder" design pattern. (The builder design pattern is one of the 23 discussed by the Gang of Four.)

A builder will take advantage of factories to create more complex objects.